### PR TITLE
[BUILD] Support passthrough env var CI into build container

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

Continuous integration(aka. CI) services like GitHub Actions, Travis always provide an environment variable `CI` in runners, and we detect this variable to run some specific actions.

In [KYUUBI #742], we add a `CI` env var detection in `build/mvn` to suppress noisy maven logs in CI scenes.

This PR introduces a new build arg `CI` in `docker/Dockerfile`, which is intent to passthrough env var CI into the build container to suppress noisy maven logs in GitHub Action and Travis.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by pan3793** [comment]
@ulysses-you Just curiosity, why `ARG kyuubi_uid=10009` is lowercase?

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -153,6 +154,7 @@ jobs:
       - name: Build Kyuubi Docker Image
         uses: docker/build-push-action@v2
         with:
+          build-args: CI="GTIHUB_ACTION_DOCKER"
```
Comment (pan3793) [inline .github/workflows/master.yml:157]: We can set everything except empty string to `CI` to suppress maven logs.

File: docker/Dockerfile [hunk 2]
```diff
@@ -28,8 +32,11 @@ ARG BASE_IMAGE=openjdk:8-jdk
 
 FROM maven:3.6-jdk-8 as builder
 
+ARG CI
 ARG MVN_ARG
 
+ENV CI ${CI}
```
Comment (yaooqinn) [inline docker/Dockerfile:38]: what’s this？

Comment (pan3793) [reply docker/Dockerfile:38]: Both GitHub Action and Travis will add `CI` into env.

https://github.com/apache/incubator-kyuubi/blob/master/build/mvn#L27-L29

Comment (ulysses-you) [reply docker/Dockerfile:38]: I think the name is not readable, can we change it to a more meaning name ? or can we use `MAVEN_CLI_OPTS` directly ?

Comment (pan3793) [reply docker/Dockerfile:38]: How about change to
```
ARG LESS_MAVEN_LOG
ENV CI ${LESS_MAVEN_LOG}
```

**Comment by yaooqinn** [comment]
the Dockerfile I made at the very beginning is release-oriented, now is dev-oriented, and after this PR is test-oriented?

BTW, as maintainers, can we put some content to describe what the PR does? If we are not following this, how can we force others to follow？

**Comment by pan3793** [comment]
> the Dockerfile I made at the very beginning is release-oriented, now is dev-oriented, and after this PR is test-oriented?

If we don't set the new build arg `CI`, it will keep the current behavior. Set build arg `CI` in GA and Travis are straightforward I think.

> BTW, as maintainers, can we put some content to describe what the PR does? If we are not following this, how can we force others to follow？

Will update soon.

File: docker/Dockerfile (hunk 2 above)
Comment (pan3793) [reply docker/Dockerfile:44]: The previous PR title and description may be misleading. Updated and please review again.

**Review by cfmcgrady — APPROVED** [review]

**Comment by cfmcgrady** [comment]
I think it's already test-oriented after [PR-955](https://github.com/apache/incubator-kyuubi/pull/955).

**Review by yaooqinn — APPROVED** [review]

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master

**Comment by ulysses-you** [comment]
I think we should consider make a new dockerfile that use to release.

**Pull request closed without merging**